plugins/sourcegraph/access_token.go

```diff
@@ -0,0 +1,63 @@
+package sourcegraph
+
+import (
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func AccessToken() schema.CredentialType {
+	return schema.CredentialType{
+		Name:    credname.AccessToken,
+		DocsURL: sdk.URL("https://docs.sourcegraph.com/cli"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Username,
+				MarkdownDescription: "Username on your Sourcegraph instance.",
+				Secret:              false,
+				Composition: &schema.ValueComposition{
+					Charset: schema.Charset{
+						Lowercase: true,
+						Digits:    true,
+						Specific:  []rune{'.', '-', '_'},
+					},
+				},
+			},
```

Comment:
Thanks for changing this, Arun 😄 However, I still think this needs some work.

Specifically, I would like to completely exclude the `username` field from this credential's definition.

This is because nor the provisioner neither the importer use this field, so functionally it serves no purpose.

That being said, there would be no problem if the user decides to add the username of their account within the item where the AccessToken and Endpoint are stored, but this should not be the responsibility of the Shell Plugin.

Let me know what you think.